src/server/worker/pipeline/transform/worker.go

```diff
@@ -374,6 +374,10 @@ type datumCallback = func(ctx context.Context, logger logs.TaggedLogger, env []s
 
 // TODO: There are way too many parameters here. Consider grouping them into a reasonable struct or storing some of these in the driver.
 func forEachDatum(ctx context.Context, driver driver.Driver, baseLogger logs.TaggedLogger, task *DatumSetTask, status *Status, cacheClient *pfssync.CacheClient, di datum.Iterator, setOpts []datum.SetOption, cb datumCallback) error {
+	jobInfo, err := driver.GetJobInfo(task.Job)
+	if err != nil {
+		return errors.Wrap(err, "could not get job info from datum set task job")
```

Comment:
include job in error - something like `errors.Wrapf(err, "load datum set's job info for job %q", task.Job.String())`